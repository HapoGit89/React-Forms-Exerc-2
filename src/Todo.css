.Todo {
    width: 35vh;
    height: 15vh;
    background-color: beige;
    border-style: solid;
    margin:2vh;
    font-size: small;
}

.TodoItem 
{
    display:flex;
    flex-direction: row;
}

.deleteButton{
    height:30px;
    width: 30px;
}