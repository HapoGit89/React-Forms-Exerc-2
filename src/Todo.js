import React from "react"
import './Todo.css'

function Todo({title, text, deleteTodo}) {
   
    return (
        <div className="TodoItem">
           <div className="Todo" >
            <h1>{title}</h1>
            <p><b>Details: </b>{text}</p>
           </div>
           <button className="deleteButton" onClick={deleteTodo}>X</button>
        </div>
    );
  }

  export default Todo
