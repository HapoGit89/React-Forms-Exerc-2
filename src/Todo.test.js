import { render, screen } from '@testing-library/react';
import App from './App';
import React from "react"
import Todo from './Todo';



// Smoke Test

it("renders without crashing", function() {
  render(<Todo text="testing" title="test" />);
});


// Snapshot Test:

it ("matches Snapshot", function() {
  const {asFragment} = render( <Todo text="testing" title="test" />);
  expect(asFragment()).toMatchSnapshot();
});


