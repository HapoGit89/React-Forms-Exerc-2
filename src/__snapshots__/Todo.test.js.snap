// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoItem"
  >
    <div
      class="Todo"
    >
      <h1>
        test
      </h1>
      <p>
        <b>
          Details: 
        </b>
        testing
      </p>
    </div>
    <button
      class="deleteButton"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
